### DISCUSSION TITLE
"mirrored" vs  "bridge" networking mode in v2.1.5?

### DISCUSSION THREAD

**Discussion by Boilerplate4u** [microsoft/WSL#11326] [General]
Hi! I'm looking for more in-depth documentation or an explanation of the internal workings of  **`"mirrored" networking mode`** in relation to both Windows and Linux. I'm currently running version 2.1.5 which came with Windows 23H2.

I used to run `bridge` mode before upgrading to 23H2 but it seems broken now. Thus, I wonder if one can accomplish the same thing using `mirrored` mode (which I'm also unable to get to work). While I'm very familiar with both Windows and Linux as well as troubleshooting network issues using tools like Wireshark and tcpdump, the lack of in-depth WSL documentation makes this a pretty challenging 'black box' scenario.

Specifically:

- Is it possible to make `mirrored` mode act like a complete transparent bridge?
- Any workarounds to make the original `bridge` networking mode work again?
- How are the network interfaces mapped between the two systems and in which end do you control capabilities like broadcasting?
- Is it possible to make the Windows network interface connected to WSL completely transparent (ie without impact of the Windows firewall)?
- Where and how (in Windows or Linux) should IP addresses be assigned using DHCP or statically?

Any ideas are welcome!

**Comment by bimmerdriver** [comment]
I would like to see more documentation on this as well. I suppose different users have different objectives, but I wish there was a networking mode where WSL acted like a completely separate host with its own IPv4 and IPv6 addresses on the network, to facilitate the use of wireshark when monitoring WSL network traffic. Having WSL embedded in the Windows host makes it difficult to isolate its network traffic.

**Comment by shigenobuokamoto** [comment]
i am not an insider. this is written based on my own analysis.

> Is it possible to make mirrored mode act like a complete transparent bridge?

can not. mirrored network is hyper-v internal switch. the only connection is to windows internal network.

> Any workarounds to make the original bridge networking mode work again?

yes, can do it. bridged network is still available. probably only bridged network will satisfy your needs.

> How are the network interfaces mapped between the two systems and in which end do you control capabilities like broadcasting?   
```
             LAN         |              Windows          |     Hyper-V switch(FSE)  |     Linux (WSL)    |
other PC                 NIC                  viertual NIC                          virtual NIC(eth0)
192.168.0.1 ---------->  192.168.0.5     -> 169.254.73.152  ----------------------> 192.168.0.5
           <----------------(relay by windows process)----------------------------- 192.168.0.5
```
169.254.73.152: this address is defined in /init. and you can find the route with the command below.

```
$ ip r list table all
127.0.0.1 via 169.254.73.152 dev loopback0 table 127 proto kernel src 127.0.0.1 onlink
192.168.0.5 via 169.254.73.152 dev eth0 table 128 proto kernel src 192.168.0.5 onlink
```

> Is it possible to make the Windows network interface connected to WSL completely transparent (ie without impact of the Windows \
firewall)?

you need to open the ports using NetFirewall or NetFirewallHyperV.

> Where and how (in Windows or Linux) should IP addresses be assigned using DHCP or statically?

there are no unique settings because mirrored network imitates the windows's ip address.

**Reply by 4-FLOSS-Free-Libre-Open-Source-Software** [reply]
> yes, can do it. bridged network is still available. probably only bridged network will satisfy your needs.

Sad news from just released [WSL v2.4.5](https://github.com/microsoft/WSL/releases/tag/2.4.5) changelog:

> Add user warning letting users know bridged networking will be deprecated

Bridged mode works very well as it is now in WSL-Version: 2.4.5.0

**Comment by jcrben** [comment]
I'm seeing a lot of dns errors - unable to resolve - and they weirdly seem to start only about 60 seconds after the vm boots up.

See also [How to use 'mirrored' mode since 'bridged' networking mode stopped working with 23H2 (WSL 2.1.5) #11291](https://github.com/microsoft/WSL/issues/11291)

No issues if I remove networkingMode=bridged and vmSwitch=myswitch1

Config below:
```
[wsl2]
networkingMode=bridged
vmSwitch=myswitch1

my /etc/wsl.conf looks like this:
benws@lenovo-legion:~$ cat /etc/wsl.conf

[boot]
systemd=true

[interop]
appendWindowsPath = false

[network]
generateResolvConf=false

benws@lenovo-legion:~$ cat /etc/resolv.conf
nameserver 1.1.1.1
#nameserver 8.8.4.4
#nameserver 8.8.8.8
```
